scripts/signal_hilight.pl

```diff
@@ -0,0 +1,134 @@
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# About
+# =====
+#
+# When you're away, sends highlighted messages to you via Signal.
+#
+#
+# To use
+# ======
+#
+# 1. Install signal-cli (https://github.com/AsamK/signal-cli).
+# 2. Register a new Signal account for this device, or link to an existing
+#    one.
+#
+#    It's better to create a dedicated Signal account to have this script
+#    send messages from. This requires a dedicated phone number to receive
+#    the verification code text message for the new Signal account. Google
+#    Voice numbers are useful here.
+#
+#    Linking gives the host running irssi the same access to the account
+#    (such as reading messages, changing account settings, and more) as the
+#    Signal app has on your phone or computer.
+#
+#    - Register a new account:
+#      - signal-cli -u +12128675309 register
+#      - Retrieve verification
+#      - signal-cli -u +12128675309 verify VERIFICATION-CODE-FROM-SMS
+#
+#    - Link to an existing account:
+#      - signal-cli link -n $(hostname)
+#      - qrencode -o device-link-qrcode.png 'tsdevice:/?uuid=...'
+#      - View device-link-qrcode.png.
+#      - Scan the QR code with the Signal app
+#        (Settings -> Linked Devices -> Link New Device).
+#
+# 3. Set $FROM and $TO, below, to the Signal phone numbers to send
+#    highlighted messages from and to.
+
+use strict;
+use warnings;
+
+use POSIX;
+
+use Irssi;
+
+our $VERSION = '1.0';
+our %IRSSI = (
+	author => 'John Morrissey',
+	contact => 'jwm@horde.net',
+	name => 'signal_hilight',
+	description => 'Send highlighted messages via Signal',
+	licence => 'BSD',
+);
+# Full phone numbers with leading + and country code.
+# For example, '+12128675309'.
+my $FROM = '';
+my $TO = '';
+
+
+my $IS_AWAY = 0;
+
+sub send_notification($) {
+	my ($message) = @_;
+
+    my $pid = fork();
+    if ($pid > 0) {
+        Irssi::pidwait_add($pid);
+    } else {
+        eval {
+            if (open(SIGNAL_CLI, '|-', "signal-cli -u $FROM send $TO")) {
+                print SIGNAL_CLI "$message";
+                close(SIGNAL_CLI);
+            }
+        };
+        POSIX::_exit(0);
+    }
+}
+
+sub sig_print_text {
+	my ($dest, $text, $stripped) = @_;
+
+	if ($IS_AWAY &&
+	    ($dest->{level} & MSGLEVEL_PUBLIC) &&
+	    ($dest->{level} & (MSGLEVEL_HILIGHT|MSGLEVEL_MSGS)) &&
+	    ($dest->{level} & MSGLEVEL_NOHILIGHT) == 0) {
+
+		send_notification($dest->{target} . ": $stripped");
+	}
+}
+
+sub sig_message_public($$$$$) {
```

Comment:
@ailin-nemui It may be useful to enable [this](http://search.cpan.org/~thaljef/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitSubroutinePrototypes.pm) critic in our config.